Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```diff
@@ -176,19 +206,24 @@
         return;
     }
 
+    if (!hasActiveUserGesture(*tab)) {
+        completionHandler(toWebExtensionError(nil, nil, @"open can only be called in response to a user action"));
+        return;
+    }
+
+    std::optional<Ref<WebExtensionSidebar>> sidebar = getOrCreateSidebar(*tab);
+
     if (!sidebar) {
-        completionHandler(makeUnexpected(sidebar.error()));
+        completionHandler(toWebExtensionError(nil, nil, @"could not get sidebar for active tab"));
```

Comment:
"active tab" is a bit of a loaded term in Web Extensions. Maybe just "tab"? Also keep in mind that "because " is prefixed to these errors, so this should be more like "a sidebar should not be found for the tab" so it flows with the full message.

Also, is this error actionable by the developer or is it more of an "unknown error"? If the later, then consider a more generic error. "an unknown error occurred".

Reply:
Also add API name for first parameter.

Reply:
That's a good point, I don't think there's much the developer could do about this error -- I'll change it to "unknown..."